flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTaskTest.java

```diff
@@ -1011,5 +1065,70 @@ protected void handleWatermark(Watermark mark) {
 			output.emitWatermark(mark);
 		}
 	}
+
+	/**
+	 * Uses to test handling the EndOfInput notification.
+	 */
+	static class TestBoundedOneInputOperator extends AbstractStreamOperator<String>
```

Comment:
As this is shared across three different tests, I would extract it to a separate file (move to upper level).